### DISCUSSION TITLE
Système de coordonnées

### DISCUSSION THREAD

**Discussion by seba1204** [photonsquid/7map#19] [Q&A] [answered]
Yo !
Avec la @7map/data on se demande quel sera le système de coordonnées de l'application.

Grâce à la méthode update() que vous (@7map/rendering ) appellerez à chaque nouvelle `frame`, on vous renvoi une liste de `mesh `représentant les objets à afficher. Sur quel repère se baseront nos `vertex` ? (sachant que nous recevons une liste de coordonnées GPS)

- Définir une origine au premier chargement de la map, à l'échelle de la map ?
- à chaque zoom, les coordonnées sont de plus en plus précises (avec plus de décimales) ?
- changer de repère à chaque zoom ?

Please help us !

Thanks 😂

**Comment by BSoDium** [comment, marked as answer]
Alors, on doit avoir un debrief la dessus avec la team ce jeudi, mais pour le moment je crois pouvoir répondre à la majorité de ces questions : 
- les vertex sont localisés par rapport à un repère **fixe** spécifique au `renderer` (cette partie est encore à réaliser)  d'où la nécessité de mapper effectivement les coordonnées GPS aux coordonnées du moteur de rendu _ie._ de la scène. La façon dont cette partie est gérée, ça je serai tenté de dire que c'est plutôt de votre ressort ( corrigez moi si je me trompe ).
- la précision des coordonnées ne doit absolument pas changer avec le zoom/dézoom, c'est simplement l'amplitude/la vitesse des déplacement de caméra qui doit être modifiée.
- j'aurai besoin d'un peu plus d'infos sur ce qu'un "repère" signifie pour vous parce que sinon je ne vois vraiment pas de raison de changer, on peut s'éloigner d'un objet sans changer de référentiel pour autant _a priori_.

**Reply by seba1204** [reply]
Pour la postérité, voici ce qui a été conclu :
- Une origine est définie pour chaque carte 
- Les coordonnées sont définies en tant qu'entier dont une unité correspond à la plus petite précision possible de la carte
- Les coordonnées GPS doivent être remapées dans ce nouveau repère

Cela posera probablement des problèmes lorsque nous voudrons implémenter la personnalisation / création des cartes, mais nous n'y sommes pas encore.
